Builds/VisualStudio2017/README.md

```diff
@@ -0,0 +1,260 @@
+  system ones.
+
+### Build Boost
+
+After [downloading boost](http://www.boost.org/users/download/) and unpacking it
+to `c:\lib\`, open a **Developer Command Prompt** for Visual Studio, change to
+the directory containing boost, then bootstrap the build tools:
+
+(As of this writing, the most recent version of boost is 1.66.0, which will
+unpack into a directory named `boost_1_66_0`. For higher versions of boost,
+adjust the directories provided in these examples as appropriate.)
+
+```powershell
+cd C:\lib\boost_1_66_0
+bootstrap
+```
+
+The rippled application is linked statically to the standard runtimes and
+external dependencies on Windows, to ensure that the behavior of the executable
+is not affected by changes in outside files. Therefore, it is necessary to build
+the required boost static libraries using this command:
+
+```powershell
+bjam -j<Num Parallel> --toolset=msvc-14.1 address-model=64 architecture=x86 link=static threading=multi runtime-link=shared,static stage
+```
+
+where you should replace `<Num Parallel>` with the number of parallel
+invocations to use build, e.g. `bjam -j4 ...` would use up to 4 concurrent build
+shell commands for the build.
+
+Building the boost libraries may take considerable time. When the build process
+is completed, take note of both the reported compiler include paths and linker
+library paths as they will be required later.
+
+### (Optional) Install CMake for Windows
+
+[CMake](http://cmake.org) is a cross platform build system generator. Visual
+Studio 2017 includes an integrated version of CMake that avoids having to
+manually run CMake, but it is undergoing continuous improvement. Users that
+prefer to use standard Visual Studio project and solution files need to install
+a dedicated version of Cmake to generate them.  The latest version can be found
+at the [CMake download site](https://cmake.org/download/). It is recommended you
+select the install option to add CMake to your path.
+
+As of this writing, the latest version of CMake for windows is 3.10.2.
+
+## Clone the rippled repository
+
+If you are familiar with cloning github repositories, just follow your normal
+process and clone `git@github.com:ripple/rippled.git`. Otherwise follow this
+section for instructions.
+
+1. If you don't have a github account, sign up for one at
+   [github.com](https://github.com/).
+2. Make sure you have Github ssh keys. For help see
+   [generating-ssh-keys](https://help.github.com/articles/generating-ssh-keys).
+
+Open the "Git Bash" shell that was installed with "Git for Windows" in the step
+above. Navigate to the directory where you want to clone rippled (git bash uses
+`/c` for windows's `C:` and forward slash where windows uses backslash, so
+`C:\Users\joe\projs` would be `/c/Users/joe/projs` in git bash). Now clone the
+repository and optionally switch to the *master* branch. Type the following at
+the bash prompt:
+
+```powershell
+git clone git@github.com:ripple/rippled.git
```

Comment:
We probably want to add the `--recursive` flag to get the submodules.

Reply:
Which components have submodules? I think after #2343 , `docca` will no longer be used. Are there others?

Reply:
Yeah, according to .gitmodules:```
[submodule "src/nudb/extras/beast"]
        path = src/nudb/extras/beast
        url = https://github.com/vinniefalco/Beast.git
[submodule "src/nudb/extras/rocksdb"]
        path = src/nudb/extras/rocksdb
        url = https://github.com/facebook/rocksdb.git
[submodule "src/nudb/doc/docca"]
        path = src/nudb/doc/docca
        url = https://github.com/vinniefalco/docca.git
```

Reply:
Are those actually needed for `nudb` though? Or just to build `nudb` examples?

Reply:
I'm not sure we need them, actually. If it works without the recursive them I'm fine leaving it off.

Reply:
My test build did not do a `--recursive` clone, so I don't think they are. I'll leave it out unless someone else discovers otherwise.